src/gamecontroller.cs

```diff
@@ -2170,5 +2202,485 @@ interface GCProductCategory
 
 		[Field ("GCProductCategoryKeyboard")]
 		NSString Keyboard { get; }
+
+		[iOS (16,0), Mac (13,0), NoWatch, TV (16,0), MacCatalyst (16,0)]
+		[Field ("GCProductCategoryHID")]
+		NSString GCProductCategoryHid { get; }
+	}
+
+	[iOS (16,0), Mac (13,0), NoWatch, TV (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface GCRacingWheel : GCDevice {
+		[Static]
+		[Export ("connectedRacingWheels")]
+		NSSet<GCRacingWheel> ConnectedRacingWheels { get; }
+
+		[Export ("acquireDeviceWithError:")]
+		bool AcquireDevice ([NullAllowed] out NSError error);
+
+		[Export ("relinquishDevice")]
+		void RelinquishDevice ();
+
+		[Export ("acquired")]
+		bool Acquired { [Bind ("isAcquired")] get; }
+
+		// [Export ("wheelInput", ArgumentSemantic.Strong)]
+		// GCRacingWheelInput WheelInput { get; }
+
+		[Export ("snapshot")]
+		bool Snapshot { [Bind ("isSnapshot")] get; }
+
+		[Export ("capture")]
+		GCRacingWheel Capture { get; }
+
+		[Notification, Field ("GCRacingWheelDidConnectNotification")]
+		NSString GCRacingWheelDidConnectNotification { get; }
+
+		[Notification, Field ("GCRacingWheelDidDisconnectNotification")]
+		NSString GCRacingWheelDidDisconnectNotification { get; }
+	}
+
+	// TODO after I figure out xcode14 chat question
+	// [iOS (16,0), Mac (13,0), NoWatch, TV (16,0), MacCatalyst (16,0)]
+	// [BaseType (typeof (GCRacingWheelInputState))]
+	// interface GCRacingWheelInput : GCDevicePhysicalInput {
+	// 	[Export ("capture")]
+	// 	GCRacingWheelInputState Capture { get; }
+
+	// 	[NullAllowed, Export ("nextInputState")]
+	// 	IGCDevicePhysicalInputStateDiff NextInputState { get; }
+	// }
```

Comment:
Please ask the question here since we have the PR.

Reply:
Inside GameController, there is a class GCRacingWheelInput that inherits from GCDevicePhysicalInput class (protocol as shown in gist).

The base class GCDevicePhysicalInput has two methods that return a certain type (IGCDevicePhysicalInputState and NSObject), and the derived class has the same method signatures that return different types (GCRacingWheelInputState and IGCDevicePhysicalInputStateDiff).

Here is a gist with cleaner information: https://gist.github.com/tj-devel709/924d2a76004d74a35e7174dcb470c64f

Is this a bug from Apple? If we have a protocol/interface, the purpose would be that these classes have methods that return a type that we specify right?